src/multisig/multisigTransactionsFactory.ts

```diff
@@ -0,0 +1,510 @@
+import {
+    AddressValue,
+    ArgSerializer,
+    BigUIntValue,
+    EndpointDefinition,
+    EndpointModifiers,
+    NativeSerializer,
+    OptionType,
+    OptionValue,
+    TokenIdentifierValue,
+    U32Value,
+    U64Type,
+    U64Value,
+    VariadicValue,
+} from "../abi";
+import { TokenComputer, TransactionsFactoryConfig } from "../core";
+import { Address } from "../core/address";
+import { Transaction } from "../core/transaction";
+import { TransactionBuilder } from "../core/transactionBuilder";
+import { SmartContractTransactionsFactory } from "../smartContracts";
+import { ProposeTransferExecuteContractInput } from "./proposeTransferExecuteContract";
+import * as resources from "./resources";
+
+interface IAbi {
+    constructorDefinition: EndpointDefinition;
+    upgradeConstructorDefinition?: EndpointDefinition;
+
+    getEndpoint(name: string): EndpointDefinition;
+}
+/**
+ * Use this class to create multisig related transactions like creating a new multisig contract,
+ * proposing actions, signing actions, and performing actions.
+ */
+export class MultisigTransactionsFactory extends SmartContractTransactionsFactory {
```

Comment:
Inheriting vs. doing "composition" - debatable; but here, indeed, using `SmartContractTransactionsFactory` as a base class seems the better choice :+1: